cmd/juju/commands/bootstrap.go

```diff
@@ -273,43 +273,80 @@ func (c *bootstrapCommand) Info() *cmd.Info {
 		},
 		Purpose: usageBootstrapSummary,
 	}
-	if details := c.configDetails(); len(details) > 0 {
-		if output, err := common.FormatConfigSchema(details); err == nil {
-			info.Doc = fmt.Sprintf("%s%s\n%s%s",
-				usageBootstrapDetailsPartOne,
-				usageBootstrapConfigTxt,
-				output,
-				usageBootstrapDetailsPartTwo)
-			return jujucmd.Info(info)
-		}
-	}
-	info.Doc = strings.TrimSpace(fmt.Sprintf("%s%s",
-		usageBootstrapDetailsPartOne,
-		usageBootstrapDetailsPartTwo))
+	configKeys := c.configDetails()
 
+	info.Doc = fmt.Sprintf("%s%s\n%s%s",
+		usageBootstrapDetailsPartOne,
+		usageBootstrapConfigTxt,
+		configKeys.Format(),
+		usageBootstrapDetailsPartTwo)
 	return jujucmd.Info(info)
 }
 
-func (c *bootstrapCommand) configDetails() map[string]interface{} {
-	result := map[string]interface{}{}
-	addAll := func(m map[string]interface{}) {
-		for k, v := range m {
-			result[k] = v
-		}
+// ConfigCategoryKeys represents the collection of keys supported by the
+// --config option during bootstrap grouped by the domain category the keys
+// apply to within Juju.
+type ConfigCategoryKeys struct {
+	// BootstrapKeys describes the set of keys supported by --config as
+	// bootstrap only config keys.
+	BootstrapKeys map[string]common.PrintConfigSchema
+
+	// ControllerKeys describes the set of keys supported by --config as
+	// configuration items that will be applied to the controller during
+	// bootstrap.
+	ControllerKeys map[string]common.PrintConfigSchema
+
+	// ModelKeys describes the set of keys supported by --config as
+	// configuration items that will be applied to the controllers model during
+	// bootstrap.
+	ModelKeys map[string]common.PrintConfigSchema
+}
+
+// Formamt is responsible for returning a formatted categorised string of all
+// the --config keys supported by bootstrap. This is used directly when
+// generating help docs.
+func (c ConfigCategoryKeys) Format() string {
+	builder := strings.Builder{}
+
+	if c.BootstrapKeys != nil {
+		fmt.Fprintln(&builder, "Bootstrap Configuration Keys:")
+		output, _ := common.FormatConfigSchema(c.BootstrapKeys)
+		fmt.Fprintln(&builder, output)
 	}
+
+	if c.ControllerKeys != nil {
+		fmt.Fprintln(&builder, "Controller Configuration Keys:")
+		output, _ := common.FormatConfigSchema(c.ControllerKeys)
+		fmt.Fprint(&builder, output)
+	}
+
+	if c.ModelKeys != nil {
+		fmt.Fprintln(&builder, "Controller Model Configuration Keys:")
```

Comment:
Model config keys can also be used with `--model-default`.

Reply:
Very true..... but we are documenting --config here. Adding that line could be confusing. Specifically we need to drop  the reference to the face that you are configuring the Controller Model or handle them as two seperate issues.

Reply:
I think we should have the list of model config keys twice, once for each flag. For --config we can call it "controller model config keys", but I think that's a bit misleading as it suggests that model config keys are of two kinds, for controller models and for non-controller models, but that's not the case (it's the same list). I think we should rather call it 'Model configurations keys (affecting the controller model)" or something like this. (Plus I'd drop the title case on the list names.)